src/ion/process.rs

```diff
@@ -728,6 +739,221 @@ impl<'a, F: Function> Env<'a, F> {
+                self.ranges[lr.index()].vreg = vreg;
+
+                // Create a new bundle that contains only this LR.
+                let new_bundle = self.create_bundle();
+                self.ranges[lr.index()].bundle = new_bundle;
+                self.bundles[new_bundle.index()].spillset = self.bundles[bundle.index()].spillset;
+                self.bundles[new_bundle.index()]
+                    .ranges
+                    .push(LiveRangeListEntry {
+                        range: cr,
+                        index: lr,
+                    });
+                new_bundles.push(new_bundle);
+
+                // If this use was a Def, set the StartsAtDef flag for
+                // the new LR. (N.B.: *not* Mod, only Def, because Mod
+                // needs an input. This flag specifically indicates
+                // that the LR does not require the value to be moved
+                // into location at start because it (re)defines the
+                // value.)
+                if u.operand.kind() == OperandKind::Def {
+                    self.ranges[lr.index()].set_flag(LiveRangeFlag::StartsAtDef);
+                }
+
+                trace!(
+                    "    -> created new LR {:?} range {:?} with new bundle {:?} for this use",
+                    lr,
+                    cr,
+                    new_bundle
+                );
+
+                // If there was any intervening range in the LR not
+                // covered by the minimal new LR above, add it to the
+                // spillset.
+                if pos > last_live_pos {
+                    let cr = CodeRange {
+                        from: last_live_pos,
+                        to: pos,
+                    };
+                    let spill_lr = self.create_liverange(cr);
+                    self.ranges[spill_lr.index()].vreg = vreg;
+                    self.ranges[spill_lr.index()].bundle = spill;
+                    new_lrs.push((vreg, spill_lr));
+                    self.bundles[spill.index()].ranges.push(LiveRangeListEntry {
+                        range: cr,
+                        index: spill_lr,
+                    });
+                    self.ranges[spill_lr.index()].bundle = spill;
+                    trace!(
+                        "    -> put intervening range {:?} in new LR {:?} in spill bundle {:?}",
+                        cr,
+                        spill_lr,
+                        spill
+                    );
+                }
+                last_live_pos = ProgPoint::before(u.pos.inst().next());
+
+                last_lr = Some(lr);
+                last_bundle = Some(new_bundle);
+                last_inst = Some(u.pos.inst());
+                last_vreg = Some(vreg);
+            }
+
+            // Clear the use-list from the original LR.
+            self.ranges[entry.index.index()].uses.clear();
+            self.ranges[entry.index.index()].uses.shrink_to_fit();
```

Comment:
Stylistically this can probably be also done with `... = Default::default()` since the new collection will have 0 capacity to start out with.

Reply:
Done!